### PR TITLE
Trunc for double

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/boost/simd/arch/common/simd/function/toint.hpp [hunk 1]
```diff
@@ -44,6 +51,46 @@ namespace boost { namespace simd { namespace ext
        return a0;
      }
    };
+
+   BOOST_DISPATCH_OVERLOAD(toint_
+                          , (typename A0, typename X)
+                          , bd::cpu_
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/toint.hpp]: this miss a is_native<X> check

File: test/function/simd/toint.saturated.cpp [hunk 2]
```diff
@@ -79,17 +79,23 @@ STF_CASE_TPL (" bs::saturated_(bs::toint) real",  STF_IEEE_TYPES)
...
   p_t v = p_t(1);
   iT iv = 1;
-  int M =  31;
+  int M =   sizeof(T)*8-1;
   for(int i=0; i < M ; i++, v*= p_t(2), iv <<= iT(1))
   {
-    STF_EQUAL(bs::saturated_(bs::toint)(v), r_t(iv));
-    STF_EQUAL(bs::saturated_(bs::toint)(-v), -r_t(iv));
+     std::cout << "i "<< i << " v " << v << " iv " << iv << " v " << v << std::endl;
+//     STF_EQUAL(bs::saturated_(bs::toint)(v), r_t(iv));
```
Comment (jfalcou) [inline test/function/simd/toint.saturated.cpp]: spurriosu comments & I/O

**Review by jfalcou — CHANGES_REQUESTED** [review]
the PR tlaks abotu trunc but no trunc files are touched ???

File: test/function/simd/toint.saturated.cpp (hunk 2 above)
Comment (jtlap) [reply test/function/simd/toint.saturated.cpp]: suppressed

File: include/boost/simd/arch/common/simd/function/toint.hpp (hunk 1 above)
Comment (jtlap) [reply include/boost/simd/arch/common/simd/function/toint.hpp]: done

**Comment by jtlap** [comment]
in fact it is toint saturated, the change of which implies modification of trunc behaviour.

**Review by jfalcou — APPROVED** [review]

**Pull request merged**